### PR TITLE
chore: upgrade flashinfer 0.4.0

### PR DESCRIPTION
**Pull request by zhyncs**
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by yzh119** [comment]
I checked the failed test, removing `.default` in `self._cached_module.plan.default(` should fix, it's no longer required.

@zhyncs

**Comment by zhyncs** [comment]
@yzh119 updated, thanks!

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/18368238157?pr=11364

File: scripts/ci/ci_install_dependency.sh [hunk 1]
```diff
@@ -74,3 +74,5 @@ fi
 # Show current packages
 $PIP_CMD list
 python3 -c "import torch; print(torch.version.cuda)"
+
+python3 -m flashinfer clear-cache
```
Comment (zhyncs) [inline scripts/ci/ci_install_dependency.sh:78]: we will delete this in one week
